Core/include/Acts/EventData/Measurement.hpp

```diff
@@ -295,21 +316,50 @@ class Measurement {
  * This encodes the source_link_t and hides it from the type generator.
  */
 template <typename source_link_t>
-struct fittable_measurement_helper {
-  template <Acts::ParID_t... pars>
+struct fittable_surface_measurement_helper {
+  template <BoundParametersIndices... pars>
   struct meas_factory {
-    using type = Measurement<source_link_t, pars...>;
+    using type = Measurement<source_link_t, BoundParametersIndices, pars...>;
   };
 
   using type =
-      typename detail::type_generator_t<meas_factory, eBoundParametersSize>;
+      typename detail::type_generator_t<BoundParametersIndices, meas_factory>;
+};
+
+template <typename source_link_t>
+struct fittable_volume_measurement_helper {
+  template <FreeParametersIndices... pars>
+  struct meas_factory {
+    using type = Measurement<source_link_t, FreeParametersIndices, pars...>;
+  };
+
+  using type =
+      typename detail::type_generator_t<FreeParametersIndices, meas_factory>;
+};
+
+/// @brief Builds a std::variant with the elements from two std::variant types
```

Comment:
This is a rather general construction. Also here, I'm not sure where it should be located. Any opinions?